spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -167,19 +168,30 @@ private static void buildWhere(PartTree tree, SpannerPersistentEntity<?> persist
 				StringJoiner andStrings = new StringJoiner(" AND ");
 
 				orPart.forEach(part -> {
-
 					String segment = part.getProperty().getSegment();
 					String tag = "tag" + tags.size();
 					tags.add(tag);
 					String andString = persistentEntity.getPersistentProperty(segment)
 							.getColumnName();
+					String sortedTag = "@" + tag;
+					if (part.shouldIgnoreCase() == IgnoreCaseType.ALWAYS) {
+						andString = "LOWER(" + andString + ")";
+						sortedTag = "LOWER(" + sortedTag + ")";
+					}
+					else if (part.shouldIgnoreCase() == IgnoreCaseType.WHEN_POSSIBLE) {
```

Comment:
If you would like the condition to be true to the actual exception message, this should be 
`else if (part.shouldIgnoreCase() != IgnoreCaseType.NEVER)`

Reply:
Yes will change for cosmetic reasons. However there are only 3 options in this enum and it's logically impossible to add another one. (always, never, sometimes cover all possible cases).